File: GXpert/GXpert.Web/Modules/Masters/AcademicYear/AcademicYearForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Masters.Forms;

[FormScript("Masters.AcademicYear")]
[BasedOnRow(typeof(AcademicYearRow), CheckNames = true)]
public class AcademicYearForm
{
    public string Name { get; set; }
    [HalfWidth]
    public DateTime StartDate { get; set; }
    [HalfWidth]
    public DateTime EndDate { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/AcademicYearForm.ts
import { StringEditor, DateEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface AcademicYearForm {
    Name: StringEditor;
    StartDate: DateEditor;
    EndDate: DateEditor;
    Description: TextAreaEditor;
}

export class AcademicYearForm extends PrefixedContext {
    static readonly formKey = 'Masters.AcademicYear';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AcademicYearForm.init)  {
            AcademicYearForm.init = true;

            var w0 = StringEditor;
            var w1 = DateEditor;
            var w2 = TextAreaEditor;

            initFormType(AcademicYearForm, [
                'Name', w0,
                'StartDate', w1,
                'EndDate', w1,
                'Description', w2
            ]);
        }
    }
}